command/agent/job_endpoint.go

```diff
@@ -1750,6 +1750,10 @@ func ApiServicesToStructs(in []*api.Service, group bool) []*structs.Service {
 			out[i].Identity = apiWorkloadIdentityToStructs(s.Identity)
 		}
 
+		if s.Weights != nil {
+			out[i].Weights = apiWorkloadWeightsToStructs(s.Weights)
+		}
```

Comment:
The function has a nil check itself so I think we can skip this conditional

```suggestion		
		out[i].Weights = apiWorkloadWeightsToStructs(s.Weights)
```